### ISSUE TITLE
'Invalid byte sequence' when trying to run Watson

### ISSUE THREAD

**Issue by apetresc** [nhmood/watson-ruby#89]
I'm on Ubuntu 13.10 with:

```
$ ruby -v
ruby 2.0.0p247 (2013-06-27 revision 41674) [x86_64-linux]
```

Whenever I try to run `watson` from any project directory, I get:

```
/home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:230:in `match': invalid byte sequence in UTF-8 (ArgumentError)
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:230:in `match'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:230:in `block in parse_file'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:225:in `each'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:225:in `each_with_index'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:225:in `parse_file'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:126:in `block in parse_dir'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:96:in `each'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:96:in `parse_dir'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:148:in `block in parse_dir'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:135:in `each'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:135:in `parse_dir'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:50:in `block in run'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:49:in `each'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/parser.rb:49:in `run'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/lib/watson/command.rb:151:in `execute'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/gems/watson-ruby-1.0.1/bin/watson:18:in `<top (required)>'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/bin/watson:23:in `load'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/bin/watson:23:in `<main>'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/bin/ruby_executable_hooks:15:in `eval'
    from /home/apetresc/.rvm/gems/ruby-2.0.0-p247/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by nhmood** [comment]
This has something to do with watson trying to parse non-text files I think.
I ran into the problem before and thought I fixed it with dfa12c3 but I guess that didn't work.

Can you try using the -f/--file on a single file (one that you know is just text) and see what happens?

**Comment by apetresc** [comment]
Yup, seems to work on a text file that I pass in with `-f`. I'm not sure which non-text file it's trying to hit when it gives that error, but I can dig in if you think it'll help.

**Comment by nhmood** [comment]
Yeah I thought that would be the case.

I wouldn't bother trying to find which file, I just need to add a better fix that forcing the encoding to UTF-8. I'll add a fix to check for UTF-8 and skip over if the file isn't.

**Comment by apetresc** [comment]
Cool. Looking forward to it :)

**Comment by hunterboerner** [comment]
I think it may be .DS_Store but the ignore in config isn't doing anything.

**Comment by apetresc** [comment]
I'm on Linux right now, so I doubt `.DS_Store` is the (only) problem. It's probably any of the dozens of binary resources in the project.

**Comment by radiospiel** [comment]
I was playing around a bit with that, and ran into the same problems on .DS_Store files, images, tars and some other build artifacts in the current dir. It would be nice if a plain watson installation would ignore common binary files, or, even, in addition all files that are either .gitignored or for other reasons not part of the current repo (i.e. being in a subrepo). I made an issue for that #98.

**Comment by nhmood** [comment]
Should be fixed in 44625bb12c816843036d57881f85766aa418a896, will close issue if someone can confirm the fix

**Comment by nhmood** [comment]
@hunterboerner You also mentioned that adding .DS_STORE to ignore in config doesn't work? I think it might have to  do with the regex used if you use .file. I think it looks for an extension if .something is given and the regex might not be correct to support nofilename and just .extension. I'll look into this

**Comment by jaseemabid** [comment]
Noticed the issue with a project with a bunch of binary files. Testing with latest master now. 

Update: Same issue on current master 48b5096

**Comment by spdaly** [comment]
I'm having the same issue on Mavericks: 

```
± % watson                                                                 !227
/Users/spdaly/.rbenv/versions/2.0.0-p247/lib/ruby/gems/2.0.0/gems/watson-ruby-1.0.3/lib/watson/parser.rb:237:in `match': invalid byte sequence in UTF-8 (ArgumentError

± % ruby -v                                                                !228
ruby 2.0.0p247 (2013-06-27 revision 41674) [x86_64-darwin13.0.2]
```

**Comment by nascob** [comment]
@spdaly Same here. Running on 10.9.0
`↪ ruby -v`
`ruby 2.0.0p247 (2013-06-27 revision 41674) [universal.x86_64-darwin13]`

`↪ watson`
`/Library/Ruby/Gems/2.0.0/gems/watson-ruby-1.0.3/lib/watson/parser.rb:237:in 'match': invalid byte sequence in UTF-8 (ArgumentError)`

**Comment by nascob** [comment]
@hunterboerner @spdaly  after updating `parser.rb` to [1f88bdb](com/hunterboerner/watson-ruby/blob/1f88bdb9849556ff5601e791ee88d59e781d4500/lib/watson/parser.rb) - running `watson` now generates a `.watsonresults` file which seems to have a bad encoding.
![File Generation](https://f.cloud.github.com/assets/1144752/1606711/92e6ddba-5455-11e3-9941-4a73b9b9c41d.png)
![Result Contents](https://f.cloud.github.com/assets/1144752/1606712/94f221be-5455-11e3-87f4-4c676b1c94be.png)

**Comment by nhmood** [comment]
@atanasbozhkov That might have to do with https://github.com/nhmood/watson-ruby/pull/111/files

**Comment by nhmood** [comment]
Should be fixed with c38b1f3 (merge of all of @hunterboerner great work!) and the gem has been updated to 1.0.4 https://rubygems.org/gems/watson-ruby
